x-pack/plugins/security_solution/server/lib/detection_engine/prebuilt_rules/logic/diff/calculation/algorithms/data_source_diff_algorithm.ts

```diff
@@ -78,8 +78,12 @@ const mergeVersions = ({
   diffOutcome,
 }: MergeArgs): MergeResult => {
   const dedupedBaseVersion = baseVersion ? getDedupedDataSourceVersion(baseVersion) : baseVersion;
-  const dedupedCurrentVersion = getDedupedDataSourceVersion(currentVersion);
-  const dedupedTargetVersion = getDedupedDataSourceVersion(targetVersion);
+  const dedupedCurrentVersion = currentVersion
+    ? getDedupedDataSourceVersion(currentVersion)
+    : currentVersion;
+  const dedupedTargetVersion = targetVersion
+    ? getDedupedDataSourceVersion(targetVersion)
+    : targetVersion;
 
```

Comment:
Let's extend the logic of the ddd function to include this ternary check:
```ts
export const getDedupedDataSourceVersion = (
  version: RuleDataSource | undefined
): RuleDataSource | undefined => {
  if (!version) {
    return version;
  }
  if (version.type === DataSourceType.index_patterns) {
    return {
      ...version,
      index_patterns: uniq(version.index_patterns),
    };
  }
  return version;
};
```
SO here we can more cleanly do:
```ts
const dedupedBaseVersion = getDedupedDataSourceVersion(baseVersion);
...etc
```